pkg/machines/scripts/create_machine.sh

```diff
@@ -26,6 +30,23 @@ handleFailure(){
 
 trap err_handler EXIT
 
+XMLS_FILE="`mktemp`"
+ROOT_PASSWORD_FILE="`mktemp`"
+USER_PASSWORD_FILE="`mktemp`"
```

Comment:
Do they ever get cleaned up? It's a bit nasty having to write the passwords into files. I recommend adding something here like

    trap "rm -f '$XMLS_FILE' '$ROOT_PASSWORD_FILE' '$USER_PASSWORD_FILE'" EXIT INT QUIT PIPE

Reply:
Sure they do, see
trap err_handler EXIT

I clean up these files in the body of the err_handler.

Reply:
/me is blind, sorry!